server/eclipse-project/src/test/java/au/org/democracydevelopers/corla/csv/DominionCVRExportParserTests.java

```diff
@@ -0,0 +1,585 @@
+      // makes it easy to match the setup scripts.
+      .withDatabaseName("corla")
+      .withUsername("corlaadmin")
+      .withPassword("corlasecret")
+      .withInitScript("corlaInit.sql");
+
+  /**
+   * Error message to match.
+   */
+  private static final String badNumsRegexp = "Unexpected or uninterpretable numbers in header:.*";
+
+  /**
+   * Blank properties for submitting to the DominionCVRExportParser instance.
+   */
+  private static final Properties blank = new Properties();
+
+  @BeforeClass
+  public static void beforeAll() {
+    postgres.start();
+    Properties hibernateProperties = new Properties();
+    hibernateProperties.setProperty("hibernate.driver", "org.postgresql.Driver");
+    hibernateProperties.setProperty("hibernate.url", postgres.getJdbcUrl());
+    hibernateProperties.setProperty("hibernate.user", postgres.getUsername());
+    hibernateProperties.setProperty("hibernate.pass", postgres.getPassword());
+    hibernateProperties.setProperty("hibernate.dialect", "org.hibernate.dialect.PostgreSQL9Dialect");
+    Persistence.setProperties(hibernateProperties);
+    Persistence.beginTransaction();
+
+  }
+
+  @AfterClass
+  public static void afterAll() {
+    postgres.stop();
+  }
+
+  /**
+   * Simple test of successful parsing of a tiny IRV test example, ThreeCandidatesTenVotes.csv.
+   * Tests that all the metadata and all the votes are correct.
+   * @throws IOException never.
+   */
+  @Test
+  public void parseThreeCandidatesTenVotesSucceeds() throws IOException {
+    testUtils.log(LOGGER, "parseThreeCandidatesTenVotesSucceeds");
+    Path path = Paths.get(TINY_CSV_PATH + "ThreeCandidatesTenVotes.csv");
+    Reader reader = Files.newBufferedReader(path);
+    County saguache = fromString("Saguache");
+
+    DominionCVRExportParser parser = new DominionCVRExportParser(reader, saguache, blank, true);
+    assertTrue(parser.parse().success);
+
+    // There should be one contest, the one we just read in.
+    List<Contest> contests = forCounties(Set.of(saguache));
+    assertEquals(1, contests.size());
+    Contest contest = contests.get(0);
+
+    // Check basic data
+    assertEquals(contest.name(), "TinyExample1");
+    assertEquals(contest.description(), ContestType.IRV.toString());
+    assertEquals(contest.choices().stream().map(Choice::name).collect(Collectors.toList()),
+        List.of("Alice","Bob","Chuan"));
+
+    // Votes allowed should be 3 (because there are 3 ranks), whereas winners=1 always for IRV.
+    assertEquals(contest.votesAllowed().intValue(), 3);
+    assertEquals(contest.winnersAllowed().intValue(), 1);
+
+    // There are 10 votes:
+    // 3 Alice, Bob, Chuan
+    // 3 Alice, Chuan, Bob
+    // 1 Bob, Alice, Chuan
+    // 3 Bob, Chuan, Alice
+    List<CVRContestInfo> cvrs = getMatching(saguache.id(),  CastVoteRecord.RecordType.UPLOADED)
+        .map(cvr -> cvr.contestInfoForContest(contest)).collect(Collectors.toList());
+    assertEquals(10, cvrs.size());
```

Comment:
If you want to simplify this, you could create a list of lists corresponding to each expected choice list, and then loop over them.
List<List<String>> listOfChoices = ....
for(int i = 0; i < listOfChoices.size(); ++i){
    assertEquals(listOfChoices.get(i), cvrs.get(i).choices());
}